### PR TITLE
options: update some descriptions and see_also fields.

### PR DESCRIPTION
**Pull request by markhpc**
These are additional updates for some descriptions and see_also tags that came up when reviewing #16489.

Signed-off-by: Mark Nelson <mnelson@redhat.com>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -3141,7 +3159,7 @@ std::vector<Option> global_options = {
   Option("bluestore_allocator", Option::TYPE_STR, Option::LEVEL_DEV)
   .set_default("bitmap")
   .add_tag("mkfs")
-  .set_description(""),
+  .set_description("Allocator to use for bluestore.  Choices are: bitmap (preferred) and stupid (only for testing)"),
```
Comment (xiexingguo) [inline src/common/options.cc]: @markhpc 
I think instead you shall do:

`set_enum_allowed({"bitamp", "stupid"})`

here..

File: src/common/options.cc [hunk 2]
```diff
@@ -3328,11 +3349,11 @@ Option("bluestore_throttle_cost_per_io_hdd", Option::TYPE_UINT, Option::LEVEL_AD
 
   Option("kstore_rocksdb_options", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("compression=kNoCompression")
-  .set_description(""),
+  .set_description("Options to pass through when RocksDB is used as the KeyValueDB for kstore."),
 
   Option("kstore_fsck_on_mount", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
   .set_default(false)
-  .set_description(""),
+  .set_description("Whether or run fsck on mount for kstore."),
```
Comment (xiexingguo) [inline src/common/options.cc]: Whether or **not**

File: src/common/options.cc [hunk 3]
```diff
@@ -3363,19 +3384,19 @@ Option("bluestore_throttle_cost_per_io_hdd", Option::TYPE_UINT, Option::LEVEL_AD
 
   Option("filestore_rocksdb_options", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("")
-  .set_description(""),
+  .set_description("Options to pass through when RocksDB is used as the KeyValueDB for filestore."),
 
   Option("filestore_omap_backend", Option::TYPE_STR, Option::LEVEL_ADVANCED)
   .set_default("rocksdb")
-  .set_description(""),
+  .set_description("The KeyValueDB to use for filestore metadata (ie omap).  Choices are: rocksdb (preferred) and leveldb"),
```
Comment (xiexingguo) [inline src/common/options.cc]: .set_enum_allowed({"rocksdb", "leveldb"})

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-02_10:30:38-rados-wip-kefu-testing-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2017-08-01_17:41:46-rados-wip-kefu-testing-distro-basic-mira/

**Comment by tchaikov** [comment]
@markhpc needs rebase

**Comment by liewegas** [comment]
@markhpc needs rebase

**Comment by markhpc** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
does not build

**Comment by markhpc** [comment]
retest this please

**Comment by liewegas** [comment]
needs rebase

File: src/common/options.cc [hunk 4]
```diff
@@ -1865,23 +1865,27 @@ std::vector<Option> get_global_options() {
...
 
     Option("osd_op_num_shards_hdd", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(5)
-    .set_description(""),
+    .set_description("")
+    .add_see_also("osd_op_num_shards"),
 
     Option("osd_op_num_shards_ssd", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(8)
     .set_description(""),
```
Comment (liewegas) [inline src/common/options.cc]: remove , here

File: src/common/options.cc [hunk 5]
```diff
@@ -3617,19 +3643,19 @@ std::vector<Option> get_global_options() {
...
     .set_default("")
-    .set_description(""),
+    .set_description("Options to pass through when RocksDB is used as the KeyValueDB for filestore."),
 
     Option("filestore_omap_backend", Option::TYPE_STR, Option::LEVEL_ADVANCED)
+    .set_enum_allowed({"rocksdb", "leveldb"})
     .set_default("rocksdb")
-    .set_description(""),
-
+    .set_description("The KeyValueDB to use for filestore metadata (ie omap).  Choices are: rocksdb (preferred) and leveldb"),
```
Comment (liewegas) [inline src/common/options.cc]: missing newline

Comment (liewegas) [reply src/common/options.cc]: can leave off the 'choices'.  antyhing beyond 1 sentence should go in long_description, but i don't think it's needed here

**Comment by batrick** [comment]
@markhpc ping again

**Comment by neha-ojha** [comment]
@markhpc needs rebase

**Comment by tchaikov** [comment]
i think the change is valuable to our user. so based on @markhpc work, i am incorporating the suggested changes into #26462.

**Pull request closed without merging**